src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -945,13 +1008,20 @@ static SFLoginOutput newSession(
             sessionId,
             commonParams);
 
-    if (consentCacheIdToken
-        && asBoolean(loginInput.getSessionParameters().get(CLIENT_STORE_TEMPORARY_CREDENTIAL))) {
-      CredentialManager.getInstance().writeIdToken(loginInput, ret);
+    if (asBoolean(loginInput.getSessionParameters().get(CLIENT_STORE_TEMPORARY_CREDENTIAL))) {
+      if (consentCacheIdToken) {
+        CredentialManager.writeIdToken(loginInput, ret);
+      }
+      if (loginInput.getOauthAccessToken() != null) {
```

Comment:
This is a good assertion, but maybe it would be better placed in `CredentialManager`? It would make the code more robust.

Reply:
I'd prefer to leave it here.